rpmlint/pkg.py

```diff
@@ -570,18 +570,23 @@ def _extract_rpm(self, dirname, verbose):
             )
             dirname = self.__tmpdir.name
             # TODO: sequence based command invocation
-            # TODO: warn some way if this fails (e.g. rpm2cpio not installed)
+            # TODO: warn some way if this fails (e.g. rpm2archive not installed)
 
             # BusyBox' cpio does not support '-D' argument and the only safe
             # usage is doing chdir before invocation.
             filename = Path(self.filename).resolve()
-            cwd = os.getcwd()
-            os.chdir(dirname)
-            command_str = f'rpm2cpio {quote(str(filename))} | cpio -id ; chmod -R +rX .'
-            stderr = None if verbose else subprocess.DEVNULL
-            subprocess.check_output(command_str, shell=True, env=ENGLISH_ENVIROMENT,
-                                    stderr=stderr)
-            os.chdir(cwd)
+            with pushd(dirname):
+                stderr = None if verbose else subprocess.DEVNULL
+                if shutil.which('rpm2archive'):
+                    stdout = subprocess.check_output(['cat', str(filename)], env=ENGLISH_ENVIROMENT,
+                                                     stderr=stderr)
```

Comment:
What's the benefit of using `cat` here instead of just directly read the file with python using `open`? maybe we can use something like:
```
with open(filename, 'rb') as rpm_file:
    subprocess.check_output('rpm2archive - | tar -xz && chmod -R +rX .', shell=True, env=ENGLISH_ENVIROMENT, stderr=stderr, stdin=rpm_file)
```

Reply:
Good point, done in 5320691